packages/terra-button/src/Button.jsx

```diff
@@ -199,8 +203,6 @@ class Button extends React.Component {
   handleMouseDown(event) {
     // Prevent button from showing focus styles when clicked
     this.shouldShowFocus = false;
-    // Wait until after onFocus has been triggered on browsers where it will get triggered for click
-    setTimeout(() => { this.shouldShowFocus = true; }, 300);
```

Comment:
A little background for this line of code...

Clicking on a button should not have the focus glow. However, we do want the focus glow to show if the focus is _programmatically_ set on the button. This line was added on #2531 to handle this.

This timeout was initially added by #2531 to allow showing the focus glow when programmatically setting focus on the button. The timeout on this line was increased to 300ms on #2740 because the onClick handler `handleClick` is added and needed to programmatically set focus on the button so that onBlur could get triggered (for FF, Safari, and IE) when the button loses focus. We set the focus in this case not to get the focus glow but. We set the focus on the button so that when it loses focus, onBlur would get called properly.

If we completely removed this time, this would break what was fixed on #2531 because after `handleMouseDown` is called, `this.shouldShowFocus` would still be false and subsequent calls to handleFocus would not be able to set the focus glow.


What I propose is to remove the 300ms and then add `this.shouldShowFocus = false;` and `setTimeout(() => { this.shouldShowFocus = true; });` in `handleClick` like this.

```
if (document.activeElement !== event.currentTarget) {
      this.shouldShowFocus = false;
      event.currentTarget.focus();
      setTimeout(() => { this.shouldShowFocus = true; });
    }
```

Reply:
@benbcai Okay that makes sense. I was setting `this.shouldShowFocus = true` in `handleBlur` since `onBlur` is called before `onFocus` in the chain of events.

I will go ahead and set `this.shouldShowFocus = true` in `handleClick`, But I am curious to know the significance of setTimeOut here, why do we need to do `setTimeout(() => { this.shouldShowFocus = true; });` instead of just `this.shouldShowFocus = true`.

Changes made here 746bdbe9423f725930e2d631bc5b336be1cd55a0

Reply:
@lokesh-0813 onFocus gets called immediately after handleMouseDown. If you just set this.shouldShowFocus = true in  handleMouseDown, onFocus will set the `focused` state to true when we don't want it to. Using setTimeout here will sets it back to true _after_ onFocus is called. You'll get a better understanding if you log out the value of this.shouldShowFocus in both onFocus and handleMouseDown and click on the button to see how the value changes.

Reply:
I have updated here 7d43f4888a82a1b871514f0e41cc5390bcd61d28

@benbcai but I'm not sure if lines 200-203 are necessary here. I was assuming `handleMouseDown's` 'eventPreventDefault()' will prevent calling `handleFocus` immediately after mouse down and instead calls `handleClick` where we set `this.shouldShowFocus=false` and fire the focus event manually.

However consoling resulted in same output even in the absence of 200 - 203 lines. Let me know if I am wrong here.